### PR TITLE
Enable detection of vehicles and pedestrians

### PR DESCRIPTION
**Pull request by aozo**
### Summary of changes
* Added config file for detecting obstacles.
* Added logs to print out the random seed that was used so that it can be saved to reproduce safety critical scenarios.
* Added a new option in the split dataset script to be able to visualize the dataset in the FiftyOne GUI which also annotates the dataset with ground truth bounding boxes.

### CODE REVIEW

File: configs/experimental/yolo_town1_obstacles.conf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+###### Control config #####
+--control=mpc
+#--lane_detection
+#--perfect_lane_detection
+--stop_for_vehicles=True
+--stop_for_people=True
+--stop_for_traffic_lights=False
+#--target_speed=40
+###### Carla config ######
+--random_seed=618549084
```
Comment (aozo) [inline configs/experimental/yolo_town1_obstacles.conf:39]: This is the seed used to reproduce the scenario where the ego car crashes into a vehicle that's stopped at a red light.

File: pylot/perception/detection/detection_operator.py [hunk 2]
```diff
@@ -255,7 +255,8 @@ def __run_yolo_model(self, image_np):
         anchors = [float(x) for x in anchors.split(',')]
         anchors = np.array(anchors).reshape(-1, 2)
 
-        boxes, classes, scores = yolo3_postprocess_np(prediction, image_shape, anchors, 3, model_input_shape, elim_grid_sense=False)
+        num_classes = len(self._coco_labels)
+        boxes, classes, scores = yolo3_postprocess_np(prediction, image_shape, anchors, num_classes, model_input_shape, elim_grid_sense=False)
```
Comment (aozo) [inline pylot/perception/detection/detection_operator.py:259]: Need to specify the number of classes based on the number of labels in the labels file.

File: pylot/perception/detection/obstacle.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 from pylot.perception.detection.utils import BoundingBox2D, BoundingBox3D, \
     get_bounding_box_in_camera_view
 
-VEHICLE_LABELS = {'car', 'bicycle', 'motorcycle', 'bus', 'truck', 'vehicle'}
+VEHICLE_LABELS = {'car', 'bicycle', 'bike', 'motobike', 'motorcycle', 'bus', 'truck', 'vehicle'}
```
Comment (aozo) [inline pylot/perception/detection/obstacle.py:7]: The CARLA dataset includes `bike` and `motobike` classes, so adding them here for Pylot to treat them as vehicles.

File: pylot/simulation/carla_operator.py [hunk 4]
```diff
@@ -45,7 +45,10 @@ def __init__(
             vehicle_id_stream: WriteStream, open_drive_stream: WriteStream,
             global_trajectory_stream: WriteStream, flags):
         if flags.random_seed:
-            random.seed(flags.random_seed)
+            seed = flags.random_seed
+        else:
+            seed = random.randint(0, 1000000000)
```
Comment (aozo) [inline pylot/simulation/carla_operator.py:50]: If the flag isn't used, we explicitly seed the TRNG so the seed can be logged.

File: configs/experimental/yolo_town1_obstacles.conf (hunk 1 above)
Comment (abrahamchan) [reply configs/experimental/yolo_town1_obstacles.conf:39]: @aozo Could you please add a brief comment to this line that summarizes this test scenario?

Comment (aozo) [reply configs/experimental/yolo_town1_obstacles.conf:39]: Done

**Comment by abrahamchan** [comment]
LGTM!

**Pull request merged**